Dockerfile

```diff
@@ -14,7 +14,7 @@ RUN go mod download
 
 # Copy the go source
 COPY cmd/main.go cmd/main.go
-COPY api/ api/
+COPY api/infrastructure api/infrastructure
```

Comment:
This just enables the repo to support multi-group instead of single group "infrastructure" as of now.
https://book.kubebuilder.io/migration/multi-group

Not sure if we'll need multi-group in future. We can also revert it back to be just single group. I ended up enabling multi-group when I was figuring out how to add new version and was hitting errors.

Reply:
Ah gotcha. I would say we can add multi-group later if needed to reduce scope of the PR and just stick to single-group for now

Reply:
Updated.